modules/unsupported/vsi/src/main/java/org/geotools/vsi/VSIReader.java

```diff
@@ -0,0 +1,65 @@
+/*
+ *    GeoTools - The Open Source Java GIS Toolkit
+ *    http://geotools.org
+ *
+ *    (C) 2021, Open Source Geospatial Foundation (OSGeo)
+ *
+ *    This library is free software; you can redistribute it and/or
+ *    modify it under the terms of the GNU Lesser General Public
+ *    License as published by the Free Software Foundation;
+ *    version 2.1 of the License.
+ *
+ *    This library is distributed in the hope that it will be useful,
+ *    but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ *    Lesser General Public License for more details.
+ */
+package org.geotools.vsi;
+
+import it.geosolutions.imageio.plugins.vrt.VRTImageReaderSpi;
+import java.io.IOException;
+import org.geotools.coverage.grid.GridCoverage2D;
+import org.geotools.coverageio.gdal.BaseGDALGridCoverage2DReader;
+import org.geotools.util.factory.Hints;
+import org.opengis.coverage.grid.Format;
+import org.opengis.coverage.grid.GridCoverageReader;
+
+/**
+ * This class can read a VSI data source and create a {@link GridCoverage2D} from the data.
+ *
+ * @author Matthew Northcott, Catalyst IT Ltd.
+ */
+public final class VSIReader extends BaseGDALGridCoverage2DReader implements GridCoverageReader {
+    private static final String worldFileExt = "";
```

Comment:
Not a big deal but I would probably rename it to WORLD_FILE_EXT since it's a static final.